swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
 class Vendor:
-    pass
+    def __init__(self, inventory=None):
+        self.inventory = inventory or []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    
+    def get_by_category(self, category):
+        items_in_category = []
+        for item in self.inventory:
+            if item.category == category:  
+                items_in_category.append(item)
+        return items_in_category
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.add(their_item)
+            self.remove(my_item)
+            friend.add(my_item)
+            friend.remove(their_item)
```

Comment:
👍 good job using the instance methods instead of append! one extra trick we could do is combine the add and remove together. If we remember, `remove` returns the item it removes, which we can use for `add`:

`self.add(friend.remove(their_item))`